src/hooks/useOrganizations.tsx

```diff
@@ -0,0 +1,60 @@
+import { personalOctokit } from 'bot/octokit'
+import { useSession } from 'next-auth/react'
+import { useEffect, useState } from 'react'
+
+const listForAuthenticatedUser = async (accessToken: string) => {
+  const octokit = personalOctokit(accessToken)
+  return (await octokit.rest.orgs.listForAuthenticatedUser()).data
+}
+
+const getOrganizationsData = async (accessToken: string) => {
+  const orgsData: {
+    data: OrgsData
+    isLoading: boolean
+    error: any
+  } = {
+    data: [],
+    isLoading: true,
+    error: null,
+  }
+
+  try {
+    const octokit = personalOctokit(accessToken)
+    const data = await octokit.rest.orgs.listForAuthenticatedUser()
+    orgsData.data = data.data
+    orgsData.isLoading = false
+  } catch (error) {
+    console.error('Error fetching organizations', { error })
+    orgsData.error = error
+    orgsData.isLoading = false
+  }
+
+  return orgsData
+}
+
+export const useOrgsData = () => {
+  const session = useSession()
+  const accessToken = session.data?.user.accessToken
+
+  const [organizationsData, setOrganizationsData] = useState<GetOrgsData>({
+    data: [],
+    isLoading: true,
+    error: null,
+  })
+
+  useEffect(() => {
+    if (!accessToken) {
+      return
+    }
+
+    getOrganizationsData(accessToken).then((orgs) => {
+      setOrganizationsData(orgs)
+    })
+  }, [accessToken])
```

Comment:
Let's shore up the loading state to this hook, that way we can properly have listeners on that items that change.

Also it would be worth moving all these fields into their own use state properties. [Here's a pretty good article on why it's bad to use objects in state](https://significa.co/blog/dangers-of-using-objects-in-usestate-useeffect-reactjs-hooks). The tldr is that they don't do deep equality checks so if we update `data`, that may not trigger a rerender because react only compares the _reference_ to the array, not the data in it. 


Let's keep `getOrganizationsData` to do one thing, get the data for organizations. 

From there our hook can look something like this.

```ts
import { personalOctokit } from 'bot/octokit'
import { useSession } from 'next-auth/react'
import { useEffect, useState } from 'react'

const getOrganizationsData = async (accessToken: string) => {
  const octokit = personalOctokit(accessToken)
  return await octokit.rest.orgs.listForAuthenticatedUser()
}

export const useOrgsData = () => {
  const session = useSession()
  const accessToken = session.data?.user.accessToken

  const [organizationData, setOrganizationData] = useState<
    Awaited<ReturnType<typeof getOrganizationsData>>['data'] | null
  >(null)
  const [isLoading, setIsLoading] = useState(true)
  const [error, setError] = useState<Error | null>(null)

  useEffect(() => {
    if (!accessToken) {
      return
    }

    setIsLoading(true)
    setError(null)

    getOrganizationsData(accessToken)
      .then((orgs) => {
        setOrganizationData(orgs.data)
      })
      .catch((error) => {
        setError(error)
      })
      .finally(() => {
        setIsLoading(false)
      })
  }, [accessToken])

  return {
    data: organizationData,
    isLoading,
    error,
  }
}
```